LeaderboardBackend/Services/Impl/AccountConfirmationService.cs

```diff
@@ -67,6 +68,38 @@ public async Task<CreateConfirmationResult> CreateConfirmationAndSendEmail(User
         return newConfirmation;
     }
 
+    public async Task<ConfirmAccountResult> ConfirmAccount(Guid id)
+    {
+        AccountConfirmation? confirmation = await _applicationContext.AccountConfirmations.Include(c => c.User).SingleOrDefaultAsync(c => c.Id == id);
+
+        if (confirmation is null)
+        {
+            return new ConfirmationNotFound();
+        }
+
+        if (confirmation.User.Role is not UserRole.Registered)
+        {
+            return new BadRole();
+        }
+
+        if (confirmation.UsedAt is not null)
+        {
+            return new AlreadyUsed();
+        }
+
+        Instant now = _clock.GetCurrentInstant();
+
+        if (confirmation.ExpiresAt < now)
```

Comment:
I would have expected greater or equal to now since it's "ExpiresAt"

Reply:
It needs to be less than [or equal to]. Times that are less than other times are in the past, so if the expiration date is less than (in the past) compared to now, then it's expired.